mat64/cholesky.go

```diff
@@ -14,13 +14,19 @@ import (
 	"github.com/gonum/matrix"
 )
 
-const badTriangle = "mat64: invalid triangle"
+const (
+	badTriangle = "mat64: invalid triangle"
+	badCholesky = "mat64: invalid Cholesky factorization"
+)
 
 // Cholesky is a type for creating and using the Cholesky factorization of a
 // symmetric positive definite matrix.
```

Comment:
Maybe we could say here that `Cholesky` must be initialized with `Factorize` and it must return true, otherwise the methods will panic. I'm now sure how to formulate it.